center/src/main/java/com/microsoft/hydralab/center/service/AgentManageService.java

```diff
@@ -128,4 +130,38 @@ public void updateAgentTeam(String teamId, String teamName){
 
          agentUserRepository.saveAll(agents);
     }
+
+    public Boolean downloadAgentConfigFile(SysUser requestor, String agentId) {
+        AgentUser agentUser = getAgent(agentId);
+        if (agentUser != null) {
+            try {
+                File agentConfigFile = File.createTempFile(
+                        "application",
+                        ".yml",
+                        new File(".\\"));
+
+                FileWriter fileWriter = new FileWriter(agentConfigFile.getAbsolutePath());
+                fileWriter.write("app:\n" +
+                        "  # register to Hydra Lab Center\n" +
+                        "  registry:\n" +
+                        "    # The server hostname:port of Hydra Lab Center. If nginx enabled, switch to port of nginx\n" +
+                        "    server: 'localhost:9886'\n" +
+                        "    # The Agent info registered in Hydra Lab Center, for instance if it's running on localhost, the URL would be: http://localhost:9886/portal/index.html#/auth\n" +
+                        "    name: " + agentUser.getName() + "\n" +
+                        "    id: " + agentUser.getId() + "\n" +
+                        "    secret: " + agentUser.getSecret() + "\n" +
+                        "    # Agent Type {1 : 1*WINDOWS + n*ANDROIDS , 2 : 1*WINDOWS+1*ANDROID , 3 : iOS}\n" +
+                        "    agent-type: 1");
+                fileWriter.flush();
+
+                if (userTeamManagementService.checkRequestorTeamAdmin(requestor, agentUser.getTeamId()) ||
```

Comment:
The file hasn't been downloaded. Maybe you can add a HttpServletResponse param and write the FileInputStream into the OutputStream of HttpServletResponse. After closing the Streams, the temp file can be deleted.
For example:
```
                ServletOutputStream out = null;
                FileInputStream in = null;
                try {
                    in = new FileInputStream(agentConfigFile);
                    response.setContentType("application/octet-stream;charset=UTF-8");
                    response.setHeader("Content-Disposition","attachment;filename=application.yml");
                    out = response.getOutputStream();
                    int len = 0;
                    byte[] buffer = new byte[1024 * 10];
                    while ((len = in.read(buffer)) != -1) {
                        out.write(buffer, 0, len);
                    }
                    out.flush();
                } catch (Exception e) {
                    e.printStackTrace();
                } finally {
                    response.flushBuffer();
                    try {
                        out.close();
                        in.close();
                        agentConfigFile.delete();
                    } catch (Exception e) {
                        e.printStackTrace();
                    }
                }
```

Reply:
An application file will be saved in the root folder, after clicking the button. Can you please check?

Reply:
Maybe the requirements are not synchronized clearly~ The application.yml file was used to deploy agents [wiki link](https://github.com/microsoft/HydraLab/wiki/Deploy-a-test-agent-service). The file needs to be downloaded in the browser and deleted after downloading.

Reply:
Now I understand, I will update the code!